src/components/Components/BuildStatusColumn.tsx

```diff
@@ -49,19 +49,17 @@ const BuildStatusColumn: React.FC<BuildStatusComponentProps> = ({ component, all
   );
 
   const pipelineRun = useLatestPipelineRunForComponent(component);
-  const status = pipelineRunFilterReducer(pipelineRun);
+  const status = pipelineRun ? pipelineRunFilterReducer(pipelineRun) : runStatus.PipelineNotStarted;
```

Comment:
when pipelinerun fetch call is inflight during initial component tab load, the `Build Starting` status shows for components that has successfully built earlier. Shall we include if pipelinerun loaded check as well to avoid it ?

Reply:
Added check below